contracts/forwarder/Forwarder.sol

```diff
@@ -59,12 +59,25 @@ contract Forwarder is IForwarder {
         _verifyAndUpdateNonce(req);
 
         bytes memory callData = abi.encodePacked(req.data, req.from);
+        uint preBalance = address (this).balance;
+        require(preBalance >= req.value, "FWD: insufficient value");
         require( gasleft()*63/64 >= req.gas, "FWD: insufficient gas" );
         // solhint-disable-next-line avoid-low-level-calls
         (success,ret) = req.to.call{gas : req.gas, value : req.value}(callData);
-        if ( address(this).balance>0 ) {
+
+        //this is a signed number (can be nagative). all params are known values,
+        // so can't overflow.
+        int excessBalance = int(address (this).balance - preBalance + req.value);
```

Comment:
I don't see a point of doing it either. If someone 'forgot' value at forwarder, why keep it there? 
What if ETH is sent BACK to forwarder as part of `execute`, do we want to disable that? 

If the problem is insufficient gas for transfer, the solution sounds to be checking we have enough gas, not adding extra logic.

Reply:
the problem is not "adding extra gas", but "making transaction atomic and predictable."
We handle `req.value` eth passed into the forwarder as part of this transaction - either in payable value, or sent inside outer transaction (as paymaster does).
We don't care for other value.